road/shapefile.py

```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+import geopandas as gpd
+from io import StringIO
+import json
+import geopandas as gpd
+import yaml
+import os
+import numpy as np
+
+
+def calculate_differences(est_df, obs_df):
+    """
+    Merges the estimated and observed dataframes and calculates various difference metrics.
+
+    Args:
+        est_df (pd.DataFrame): DataFrame containing estimated data with columns 'A', 'B', and time periods.
+        obs_df (pd.DataFrame): DataFrame containing observed data with columns 'A', 'B', and time periods.
+
+    Returns:
+        pd.DataFrame: Merged DataFrame with additional columns for ratio, difference, percentage difference,
+                      absolute difference, and root mean squared error (RMSE) for each time period.
+    """
+    time_periods = ['AM', 'MD', 'PM', 'EV', 'EA', 'Daily']
+    merged_df = pd.merge(est_df, obs_df, on=['A', 'B'], suffixes=('_est', '_obs'))
+
+    for period in time_periods:
+        est_col = f'{period}_est'
+        obs_col = f'{period}_obs'
+        merged_df[f'{period}_ratio'] = merged_df[est_col] / merged_df[obs_col]
+        merged_df[f'{period}_diff'] = merged_df[est_col] - merged_df[obs_col]
+        merged_df[f'{period}_pctdiff'] = (merged_df[f'{period}_diff']) / merged_df[obs_col]
+        merged_df[f'{period}_absdiff'] = abs(merged_df[f'{period}_diff'])
+        merged_df[f'{period}_rmse'] = np.sqrt((merged_df[f'{period}_diff']) ** 2)
+
+    merged_df['AB'] = merged_df['A'].astype(str) + ' ' + merged_df['B'].astype(str)
+    merged_df.to_csv('map_data.csv',index=False)

```

Comment:
maybe move this output filename to the function arg list (but it can have a default value of 'map_data.csv' maybe)?